### DISCUSSION TITLE
How can I use the C++ header #include <bits/stdc++.h> on Mac?

### DISCUSSION THREAD

**Discussion by Significant36** [cpeditor/cpeditor#1172] [Q&A] [answered]
My friend convinced me to switch to CPEditor as my IDE to use in contests. However, testing my code 

```c++
#include <bits/stdc++.h>

using namespace std;

int main(){
    cout << "Hello World" << endl;
}
```

produces errors from the compiler. If I were to switch to using for example, `<iostream>`, then it works fine. However, I don't want to include a massive number of `#include <blahblah>` before my code. 

When I check for my gcc compiler version it gives me:
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: x86_64-apple-darwin21.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

My OS version is 12.6.5.

**Comment by ouuan** [comment]
Perhaps CP Editor is not using the correct compiler. You can try setting the complete absolute path to the compiler in CP Editor settings.

**Reply by Significant36** [reply]
Sorry, I'm not quite familiar with the CP Editor settings could you guide me on how to do so?

**Reply by ouuan** [reply]
Click Options->Preferences or press <kbd>Ctrl+P</kbd>, and then navigate to C++ Commands.

**Reply by Significant36** [reply]
Thanks, I assume it's the "C++ Executable File Path". What should I put there?

**Reply by ouuan** [reply]
> I assume it's the "C++ Executable File Path"

No, it's the compile command. You can hover on them to see the tooltips or read the documentation, or guess from the default values. Are the setting names that confusing 🤔?

Do you know how to compile C++ in command line? I was suggesting that you could try replacing `c++` in the compilation command with the absolute path to the compiler.

**Reply by Significant36** [reply]
Sorry, oops I don't usually dabble with this stuff, apologies in advance.
I've never compiled C++ in the command line before, and are you suggesting I replace `c++` with  /Library/Developer/CommandLineTools/usr/bin? 
After doing so, the compiler says 
"[Failed to start compilation: Failed to start the compiler. Please check [Preferences->Language->C++->C++ Commands->C++ Compile Command or add the compiler in the PATH environment variable.]"

**Reply by ouuan** [reply, marked as answer]
The path to the compiler should be `Library/Developer/CommandLineTools/usr/bin/g++`.

**Reply by Significant36** [reply]
Alright, I've changed 'c++' with '/Library/Developer/CommandLineTools/usr/bin/g++' and I tried compiling but the compiler said "[Error occurred while compiling]" along with some other error messages.

**Reply by Significant36** [reply]
Hi, @ouuan, I figured out my problem and my compiler works normally now.

Thanks for the help!
多谢多谢
